### PR TITLE
deploy: update deploy cluster role

### PR DESCRIPTION
**Pull request by lingsamuel**
Signed-off-by: Ling Samuel <lingsamuelgrace@gmail.com>

Add missing gateway API  cluster role rules

### Type of change:

- [x] Bugfix

### CODE REVIEW

File: samples/deploy/rbac/apisix_view_clusterrole.yaml [hunk 1]
```diff
@@ -20,150 +20,161 @@ kind: ClusterRole
...
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - networking.k8s.io
+    resources:
+      - ingresses
+      - ingresses/status
+      - networkpolicies
```
Comment (tao12345666333) [inline samples/deploy/rbac/apisix_view_clusterrole.yaml:128]: we don't need `networkpolicies` resource.

File: samples/deploy/rbac/apisix_view_clusterrole.yaml [hunk 2]
```diff
@@ -20,150 +20,161 @@ kind: ClusterRole
...
+      - poddisruptionbudgets
+    verbs:
+      - get
+      - list
+      - watch
+  - apiGroups:
+      - networking.k8s.io
+    resources:
+      - ingresses
+      - ingresses/status
```
Comment (tao12345666333) [inline samples/deploy/rbac/apisix_view_clusterrole.yaml:127]: I think there should be a split here

```
  - apiGroups:
      - networking.k8s.io
    resources:
      - ingresses
    verbs:
      - get
      - list
      - watch
```

and

```
  - apiGroups:
      - networking.k8s.io
    resources:
      - ingresses/status
    verbs:
      - update

```

File: samples/deploy/rbac/apisix_view_clusterrole.yaml [hunk 3]
```diff
@@ -20,150 +20,161 @@ kind: ClusterRole
...
+      - gateway.networking.k8s.io
+    resources:
+      - httproutes
+      - tlsroutes
+      - gateways
+      - gatewayclasses
+    verbs:
+      - get
+      - list
+      - watch
```
Comment (tao12345666333) [inline samples/deploy/rbac/apisix_view_clusterrole.yaml:180]: maybe we need `status` sub-resource also.

```
- apiGroups:
  - gateway.networking.k8s.io
  resources:
  - gatewayclasses/status
  - gateways/status
  - httproutes/status
  - tlsroutes/status
  verbs:
  - update

```

File: samples/deploy/rbac/apisix_view_clusterrole.yaml [hunk 4]
```diff
@@ -20,150 +20,176 @@ kind: ClusterRole
...
+    resources:
+      - events
+    verbs:
+      - "*"
+  - apiGroups:
+      - ""
+    resources:
+      - configmaps
+      - endpoints
+      - persistentvolumeclaims
```
Comment (tao12345666333) [inline samples/deploy/rbac/apisix_view_clusterrole.yaml:34]: We don't need this one.

File: samples/deploy/rbac/apisix_view_clusterrole.yaml [hunk 5]
```diff
@@ -20,150 +20,176 @@ kind: ClusterRole
...
+      - pods
+      - replicationcontrollers
+      - replicationcontrollers/scale
+      - serviceaccounts
```
Comment (tao12345666333) [inline samples/deploy/rbac/apisix_view_clusterrole.yaml:38]: Ditto

File: samples/deploy/rbac/apisix_view_clusterrole.yaml [hunk 6]
```diff
@@ -20,150 +20,176 @@ kind: ClusterRole
...
+      - namespaces/status
+      - pods/log
+      - pods/status
+      - replicationcontrollers/status
+      - resourcequotas
+      - resourcequotas/status
+    verbs:
+      - get
+      - list
+      - watch
```
Comment (tao12345666333) [inline samples/deploy/rbac/apisix_view_clusterrole.yaml:59]: Ditto

File: samples/deploy/rbac/apisix_view_clusterrole.yaml [hunk 7]
```diff
@@ -20,150 +20,176 @@ kind: ClusterRole
...
+  - apiGroups:
+      - batch
+    resources:
+      - cronjobs
+      - jobs
+    verbs:
+      - get
+      - list
+      - watch
```
Comment (tao12345666333) [inline samples/deploy/rbac/apisix_view_clusterrole.yaml:99]: Ditto

File: samples/deploy/rbac/apisix_view_clusterrole.yaml [hunk 8]
```diff
@@ -20,150 +20,176 @@ kind: ClusterRole
...
+  - apiGroups:
+      - policy
+    resources:
+      - poddisruptionbudgets
+    verbs:
+      - get
+      - list
+      - watch
```
Comment (tao12345666333) [inline samples/deploy/rbac/apisix_view_clusterrole.yaml:122]: Ditto

File: samples/deploy/rbac/apisix_view_clusterrole.yaml [hunk 9]
```diff
@@ -20,150 +20,176 @@ kind: ClusterRole
...
+    verbs:
+      - update
+  - apiGroups:
+      - metrics.k8s.io
+    resources:
+      - pods
+    verbs:
+      - get
+      - list
+      - watch
```
Comment (tao12345666333) [inline samples/deploy/rbac/apisix_view_clusterrole.yaml:144]: ditto

**Review by tokers — APPROVED** [review]

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**